crates/ruff_dev/src/format_dev.rs

```diff
@@ -239,78 +261,88 @@ fn format_dev_multi_project(args: &Args) -> anyhow::Result<bool> {
+                info!(
+                    parent: None,
+                    file_count = result.file_count,
+                    similarity_index = result.statistics.similarity_index(),
+                    duration = result.duration.as_secs_f32(),
+                    "Finished {}: {} files, similarity index {:.3}, {:.2}s",
+                    project_path.display(),
+                    result.file_count,
+                    result.statistics.similarity_index(),
+                    result.duration.as_secs_f32(),
+                );
+                if result.error_count() > 0 {
+                    error!(
+                        parent: None,
+                        "{}",
+                        result.display(args.format).to_string().trim_end()
                     );
-                    println!("{}", result.display(args.format).to_string().trim_end());
-                });
+                }
                 if let Some(error_file) = &mut error_file {
                     write!(error_file, "{}", result.display(args.format)).unwrap();
                     error_file.flush().unwrap();
                 }
 
-                bar.inc(1);
+                pb_span.pb_inc(1);
             }
             Err(error) => {
-                bar.suspend(|| println!("Failed {}: {}", project_path.display(), error));
-                bar.inc(1);
+                error!(parent: None, "Failed {}: {}", project_path.display(), error);
+                pb_span.pb_inc(1);
             }
         }
-        bar.finish_and_clear();
     }
 
+    drop(pb_span_enter);
+    drop(pb_span);
+
     let duration = start.elapsed();
 
-    #[allow(clippy::print_stdout)]
-    {
-        println!(
-            "{total_errors} stability errors in {total_files} files in {}s",
-            duration.as_secs_f32()
-        );
-    }
+    info!(
+        parent: None,
+        "{total_errors} stability errors in {total_files} files in {}s",
+        duration.as_secs_f32()
+    );
 
     Ok(total_errors == 0)
 }
 
+#[tracing::instrument]
 fn format_dev_project(
     files: &[PathBuf],
     stability_check: bool,
     write: bool,
-    progress_bar: bool,
 ) -> anyhow::Result<CheckRepoResult> {
     let start = Instant::now();
 
     // TODO(konstin): The assumptions between this script (one repo) and ruff (pass in a bunch of
     // files) mismatch.
     let options = BlackOptions::from_file(&files[0])?.to_py_format_options();
-    debug!("Options for {}: {options:?}", files[0].display());
+    debug!(
+        parent: None,
+        "Options for {}: {options:?}",
+        files[0].display()
+    );
 
     // TODO(konstin): black excludes
 
     // Find files to check (or in this case, format twice). Adapted from ruff_cli
     // First argument is ignored
     let paths = ruff_check_paths(files)?;
 
-    let bar = progress_bar.then(|| ProgressBar::new(paths.len() as u64));
+    let pb_span = info_span!("format_dev_project progress bar", first_file = %files[0].display());
```

Comment:
Nit: You could wrap line 341..398 in a block, this way you don't need the explicit drop.